---
title: "Effects of progressive resistance training on the expression of long non-coding RNAs in skeletal muscle of young adults"
format:
  docx:
    reference-doc: resources/custom-reference-doc.docx
author: 
  - "Chidimma Phoebe Echebiri^a^"
  - "H&aring;vard Hamarsland^b^"
  - "Rafi Ahmad^a^"
  - "Stian Ellefsen^b^"
  - "Daniel Hammarstr&ouml;m^b^"
csl: resources/rna-biology-csl.csl
bibliography: resources/ref-lib.bib
prefer-html: true
---



<!-- target journal: https://www.tandfonline.com/journals/krnb20 --> 
<!-- Style requirements: This Journal allows Format-Free Submission. You don't need to format your manuscript to meet any special style requirements, but your submission should be easy to read and comment on. Your references can be in any scholarly citation format, so long as formatting is consistent throughout the article. -->
<!-- See: https://www.tandfonline.com/action/authorSubmission?show=instructions&journalCode=krnb20#Style_requirements  -->


::: {custom-style="author-meta"}
^a^ Department of Biotechnology, Inland Norway University of Applied Sciences

^b^ Section for Health and Exercise Physiology, Department of Public Health and Sport Sciences, Inland Norway University of Applied Sciences, Lillehammer, Norway


Correspondance: Chidimma Phoebe Echebiri (chidimma.echebiri@inn.no), Daniel Hammarstr&ouml;m (daniel.hammarstrom@inn.no).

:::

::: {custom-style="abstracttitle"} 
Abstract 
:::

::: {custom-style="Abstract"}

:::



```{r, setup, include=FALSE}


knitr::opts_chunk$set(warning = FALSE, message = FALSE)

```

```{r, libraries , echo=FALSE}

source("R/Trainome_functions.R")
library(ggplot2)
library(dplyr)
library(tidyverse)
library(RColorBrewer)
library(ggrepel)
library(lme4)
library(lmerTest)
library(gridExtra)
library(clusterProfiler)
library(org.Hs.eg.db)


#Write the codes to be called in the script
df <- readRDS("data/seqwrap_generated_models/filtered_training_model.RDS")

#DE at mid exercise
t3 <- df %>%
  dplyr::filter(coef == "timet3")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1) %>%
  dplyr::select(target,adj.p, log2fc, pval.unif, pval.disp )



#DE midexercise with training interaction
t3_int <- df %>%
  dplyr::filter(coef == "training_statustrained:timet3")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)%>%
  dplyr::select(target,adj.p, log2fc, pval.unif, pval.disp )


# DE postexercise
t4 <- df %>%
  dplyr::filter(coef == "timet4")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)%>%
  dplyr::select(target,adj.p, log2fc, pval.unif, pval.disp )


#Interaction with training
t4_int <- df %>%
  dplyr::filter(coef == "training_statustrained:timet4")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)%>%
  dplyr::select(target,adj.p, log2fc, pval.unif, pval.disp )



#Load the mid exercise training model for correlation testing
train_cor <- readRDS("data/seqwrap_generated_models/training_coexpression_models/mid_exc_correlation_counts.RDS")



#The post exercise correlation model
train_post <- readRDS("data/seqwrap_generated_models/training_coexpression_models/post_exc_correlation_counts.RDS")


genes_fpkm <- readRDS("data/Ct_genes_FPKM.RDS")

```

## Abstract

The benefits of progressive resistance exercise training (RT) are largely known and relatively well-researched. RT is a potent way of stimulating an increase in muscle strength and mass which contribute to an individual's quality of life. Loss of muscle strength and mass tend to occur as a result of aging and certain disease conditions. The challenge often lies in personalizing an individual's regimen as different individuals respond differently to the same RT program. As per today, a one-size-fits-all approach is in use. This means it is necessary to not only understand the responses to RT, but also understand what regulates those responses. Lon non-coding RNAs (lncRNAs) are said to play regulatory roles in different aspects of cellular physiology. Most studies today are focused on their roles in diseases, especially cancers. This study is focused on understanding the effects of RT on the expression of long non-coding RNAs in skeletal muscle of young adults; as a step towards understanding how responses to RT are regulated. lncRNAs expressed in the *vastus lateralis* muscle of young adults were modeled using glmmTMB to analyse the effects of RT in trained versus untrained legs, and the effects of volume of RT on trained legs. lmer was used to build a model that predicted protein-coding genes that are coexpressed with the differentially expressed lncRNAs based on the correlation of their expression patterns.

## Introduction

<!-- Proposed intro -->

The skeletal muscle system plays crucial roles in health maintenance across the lifespan, including regulating whole-body energy homeostasis and enabling physical independence. Therefore, it serves as a critical therapeutic target for overall health promotion, with physical exercise being the most effective non-pharmacological intervention to promote skeletal muscle health. More specifically, progressive resistance training (RT) is known to increase skeletal muscle mass in a volume-dependent manner [@hammarstrom_benefits_2020] and has been shown to improve multiple indices of health in various populations, such as reducing blood glucose levels [@xie_effects_2022; @westcott_resistance_2012] and improving strength and performance in older adults [@liu_progressive_2009] and in children with cerebral palsy [@hanssen_progressive_2022], improve health status and pain intensity in women with fibromyalgia [@larsson_resistance_2015; @busch_resistance_2013], counteract the adverse effects of androgen deprivation therapy in prostate cancer patients [@houben_resistance_2023], as well as decrease menopause-related symptoms [@berin_resistance_2019].

Mechanistically, skeletal muscles adapt to repeated RT-induced stress with increased quality and mass through several parallel processes, including the incorporation of muscle-specific stem cells into muscle cells, a transient increase in ribosomal biogenesis[@hammarstrom_ribosome_2022] and protein synthesis, which promotes muscle hypertrophy, structural adaptation in connective tissues and an altered muscle phenotype. Although selected components of the adaptive response to RT are well-described, regulatory mechanisms, such as roles of non-protein-coding RNA, including long non-coding RNA (lncRNAs), are much less understood in the context of skeletal muscle adapting to RT. 


- Differential expressed lncRNAs
- Infer roles by correlating with protein coding RNA


<!-- End proposed intro --> 



The skeletal muscle is important for physical health and vitality. It responds to both use and disuse by hypertrophying, and loss of muscle strength and mass respectively [@lim_evidence-based_2022; @khan_increased_2020 ]. Aging in humans correlate to loss of muscle mass and strength [@power_human_2013; @volpi_muscle_2004 ] which in turn correlate to the risks of falls, injuries and impaired mobility @wolfson_strength_1995 . Certain disease conditions like chronic obstructive pulmonary disease (COPD) could also accelerate loss in muscle mass and strength @molmen_chronic_2021 . Thus, muscle strength and mass are important for improved quality of life.

Progressive resistance exercise training (RT); a type of exercise where the skeletal muscle is exercised against progressively increased types of resistance @liu_progressive_2009 such as free weights , is the most potent non-pharmacological method of stimulating muscle hypertrophy and countering the loss in muscle strength and mass [@lim_evidence-based_2022; @hurst_resistance_2022] . RT has been shown to reduce blood glucose levels [@xie_effects_2022; @westcott_resistance_2012] improve strength and performance in older adults @liu_progressive_2009 and in children with cerebral palsy @hanssen_progressive_2022 , improve health status and pain intensity in women with fibromyalgia [@larsson_resistance_2015; @busch_resistance_2013] , counteract the adverse effects of androgen deprivation therapy in prostate cancer patients @houben_resistance_2023 , as well as decrease menopause-related symptoms @berin_resistance_2019. The benefits of RT are dependent on its volume @hammarstrom_benefits_2020 and extend beyond physical health to include the mental health of participants @kekalainen_effects_2018 . Besides volume, factors such as ribosome biogenesis, transcriptome profile and responses, metabolites profile *etc* determine an individual's response to RT [@lim_evidence-based_2022; @hammarstrom_ribosome_2022].

Numerous studies have reported the gene expression patterns following RT, but how these patterns are regulated is poorly understood. Understanding these could also help understand the factors regulating response to RT. Long non-coding RNAs (lncRNAs) are said to regulate the expression of protein-coding genes [@statello_gene_2021; @doi:10.1177/2472630318780639] as well as different aspects of cellular physiology and function [@oo_long_2022; @bridges_lnccation_2021] . lncRNAs are non-protein-coding RNAs of more than 200 nucleotides in length. Most lncRNAs have low levels of expression and sequence conservation @mattick_long_2023 , their expressions however are highly tissue and condition-specific [@statello_gene_2021; @bonilauri_long_2020; @ali_beyond_2020] . Over 240,000 lncRNAs in humans have been curated @10.1093/nar/gky960 even though not much is known about the functional roles of a majority of them @ali_beyond_2020 . Some have been reported to play roles in cell cycle regulation [@kitagawa_cell_2013; @khan_distinct_2023] , differentiation @delas_lncrna_2017 , metabolism @sirey_long_2019 , and muscle regeneration @sweta_importance_2019 . There is currently significant uncertainty about the mechanisms of actions of most lncRNAs. Figuring out lncRNAs that are differentially expressed in a given condition could thus be the first in a series of steps aimed towards elucidating their functions and mechanisms of action.

Most publicly available research on this group of RNAs are focused on their roles in diseases, especially cancers, . Relatively fewer studies like @bonilauri_long_2020 are available on their roles in health promotion. It is perhaps pertinent to investigate the possible functional roles of lncRNAs in health promoting activities like RT. This knowledge could not only contribute towards a deeper understanding of the mechanisms of response to RT, but perhaps optimizing the benefits as well as personalizing an individual's RT regimen. This knowledge could also be extended to other areas of health where specific RNAs have been confirmed to play roles in diseases. lncRNAs have a strong potential of usage as biomarkers of diseases, or as therapeutic agents @correia2021 , as such they could be great tools for personalised medicine.

This study aims to identify lncRNAs that are differentially expressed following RT based on exercise conditions, that is, between the trained and untrained legs of participants, and between RT volumes among the trained. The study hopes to enhance the current knowledge about the responses to RT and how these can be optimized and personalized for individual participants.

## Methods

### Study design, RNA extraction and sequencing

Study design , sample preparation and RNA sequencing were done as described by Hamarsland et al (in_view).



### Data Analyses

The quality of the RNA sequences was checked using FastQc (v0.11.5) @andrews2012 while RSEM (v.1.3.3-foss-2019b) @li2011 was used to align and count the reads mapping to human genome(GRCh 38) with gencode v40 primary assembly annotation . The expected count and FPKM values of the gene level counts were used during the data analyses . Further data extraction and statistical analyses were done on RStudio (2023.06.2Â Build 561). An overview of data analyses steps is shown in Figure 1

#### Models and visualization

Using BiomaRt @durinck_mapping_2009 , all genes annotated as lncRNA by the Ensembl database were filtered for downstream analyses. A total of 1024 lncRNAs were included in the analyses. These lncRNA counts were modeled using negative binomial GLMM (Generalized Linear Mixed Model) with the effective library size and time as fixed effects. Each participant was used as random effect having an individual intercept. To analyse the effect of RT over time, differential expression of lncRNAs between trained and the untrained contralateral legs was evaluated with the interaction between training status and time added to the fixed effects. To analyse the volume-dependent effects of RT on lncRNA expression, the interaction between the different training conditions and time were added to the fixed effects. Both model types were iteratively fitted using glmmTMB in Seqwrap , an R package built by Hammastrom et al (in view). All models were evaluated for uniformity and dispersion using DHARma (0.4.6). lncRNAs were identified as differentially expressed when their absolute log2 fold change was greater than 1 or less than -1, their adjusted p-values equal to or less than 0.05, and the p values of their model's uniformity and dispersion test above 0.05. All the DE lncRNAs based on interaction effects of training status, and RT conditions were individually researched in publicly available literature to ascertain current knowledge about them

#### Coexpression analyses

To predict the potential functions of the differentially expressed lncRNAs, coexpressed protein-coding genes were modeled and iteratively fitted using lmer @bates_fitting_2015 in Seqwrap . Each lncRNA count, time, condition, and sex of participants were used as fixed effects, while each participant was used as random effect with a fixed intercept. Coexpressed proteins were identified as those with adjusted p values above or at 0.05 and p values of their DHaRma uniformity test above 0.05. Gene ontology and KEGG gene sets of the coexpressed protein-coding genes were retrieved from org.Hs.eg.db.

## Results and Discussion

This work was aimed at investigating which lncRNAs were differentially expressed in young individuals following about 12 weeks of RT.

#### Trained versus untrained

15 lncRNAs were differentially expressed (DE) between the trained and untrained legs at midexercise , while only eight were differentially expressed by the end of the training regimen see Tables 1 and 2.

```{r, DE_midexercise,tab.cap = "Differentially expressed lncRNAs at mid exercise",   echo=FALSE}

knitr::kable(t3, caption ="Differentially expressed lncRNAs halfway through the training program" )

knitr::kable(t3_int, caption ="Differentially expressed lncRNAs halfway through the training program among trained legs" )



```

The interaction effect between the training status of the participant's leg and time showed 15 DE lncRNAs at midexercise and 1 at postexercise as shown in the Tables 3 and 4.

```{r,DE_postexercise,  echo=FALSE}


knitr::kable(t4, caption = "DE lncRNAs between the trained and untrained legs postexercise" )


knitr::kable(t4_int, caption = "DE lncRNAs between the trained and untrained legs postexercise among trained legs" )


```

While the role of many lncRNAs are yet to be understood, some of the DE lncRNAs had been reported to play roles in certain disease conditions, especially cancers.

##### Differentially expressed lncRNAs in trained legs at midexercise

The DEs of interest are those who expression patterns are results of the interaction between training status and time. That is, they show the DE lncRNAs among trained legs. see Figure 1.

```{r, Volcano_plot_midexercise,fig.cap= "Volcano plot of DE lncRNAs at midexercise", fig.align='left', echo=FALSE}
t3_int <- df %>%
  dplyr::filter(coef == "training_statustrained:timet3")
plot_volcano(t3_int, "DE lncRNAs between the trained and untrained at midexercise")


```

The MEF2C-AS1 was up-regulated at midexercise in trained legs . It is reportedly down-regulated significantly in gastric cancer @luo2018 and cervical cancer \[ @guo2022; @wang2021 \]. It plays tumour suppressor roles during tumorigenesis @qian2022. Its over-expression suppressed cervical cancer @wang2021 which suggests that its up-regulation following a few weeks of RT could potentially be beneficial to health. MEF2C-AS1 alongside five other lncRNAs could predict prognosis in breast cancer patients. @luo2022. ARRDC3-AS1 was upregulated among trained legs but was downregulated in lung cancer @shah2024 @yu2020. LINC00702 was quite lowly expressed in colorectal cancer tissues @yu2020 and in lung cancer @yu2019 but is upregulated in trained legs at midexercise. PKN2-AS1 is over-expressed in sepsis @yuan2022 but down-regulated by RT. Overexpression of NPSR1-AS1 is a marker of shorter disease-specific survival or overall survival in lung adenocarcinoma @zhang2022 and promotes the proliferation and glycolysis of hepatocellular carcinoma cells @he2020; it was down-regulated by RT The current knowledge about the lncRNAs and their expression patern following RT tends to suggest that RT might have some disease-suppression or preventing effects.

719 protein-coding genes were correlated to their expression. The gene ontology results of the coexpressed proteins are as shown in \ref(fig:Correlation_plot_bp), \ref(fig:Correlation_plot_mf) and \ref(fig:Correlation_plot_cc)

```{r, Correlation_plot_cc,fig.align='left', echo=FALSE}


ego_df <- enrichGO(gene = train_cor$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "cc",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df, showCategory = 15,
        
        font.size = 5, title = "15 top cellular components of coexpressed proteins at  midexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8), plot.title = element_text(hjust = 0.5, size = 8))

```

```{r, correlation_plot_bp, fig.align='right', echo=FALSE}
ego_df_2 <- enrichGO(gene = train_cor$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "BP",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df_2, showCategory = 15,
        
        font.size = 5, title = "15 top Biological processes of coexpressed proteins at  midexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8),  plot.title = element_text(hjust = 0.5, size = 8))

```

```{r, correlation_plot_mf, echo=FALSE}
ego_df_3 <- enrichGO(gene = train_cor$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "MF",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df_3, showCategory = 15,
        
        font.size = 5, title = "15 top Molecuar functions of coexpressed proteins at  midexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8),  plot.title = element_text(hjust = 0.5, size = 8))

```

```{r, gene_enrichemnt_mid, echo=FALSE}
entrez_ids <- bitr(train_cor$geneid, "SYMBOL", "ENTREZID", org.Hs.eg.db)


#pathway overrepresentation analyses
kegg_df <- enrichKEGG(gene = entrez_ids$ENTREZID,
                      organism = "hsa",
                      keyType = "kegg",
                      # OrgDb = org.Hs.eg.db, 
                      #ont = "MF", 
                      pAdjustMethod = "BH", 
                      qvalueCutoff = 0.05)


barplot(kegg_df, showCategory = 15, title = "10 most enriched pathways in co-expressed protein coding genes among trained indivuduals at midexercise")+
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8),  plot.title = element_text(hjust = 0.5, size = 8))

```

##### Differentially expressed lncRNAs in trained legs post exercise

DUXAP9 was down-regulated in the trained leg postexercise. Its over-expression has been linked to different cancers [@zhu2018; @chen2019; @zhu2019] . DUXAP9's expression according to the model, correlated with the expression of 184 protein-coding genes. see Figures below.

```{r, Volcano_plot_postexercise,fig.cap= "Volcano plot of DE lncRNAs postexercise", fig.align='left', echo=FALSE}
t4_int <- df %>%
  dplyr::filter(coef == "training_statustrained:timet4")
plot_volcano(t4_int, "DE lncRNA between the trained and untrained at postexercise")
```

The gene ontology and Kegg enrichment plots are as shown in

```{r, correlations_postexercise, echo=FALSE}


ego_df_4 <- enrichGO(gene = train_post$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "MF",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df_4, showCategory = 15,
        
        font.size = 5, title = "15 top Molecuar functions of coexpressed proteins at  postexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8), , plot.title = element_text(hjust = 0.5, size = 8))


ego_df_5 <- enrichGO(gene = train_post$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "BP",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df_5, showCategory = 15,
        
        font.size = 5, title = "15 top Biological processes of coexpressed proteins at  postexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8),  plot.title = element_text(hjust = 0.5, size = 8))


ego_df_6 <- enrichGO(gene = train_post$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "cc",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(ego_df_6, showCategory = 15,
        
        font.size = 5, title = "15 top cellular components of coexpressed proteins at  postexercise of trained legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8),  plot.title = element_text(hjust = 0.5, size = 8))



entrez_ids <- bitr(train_post$geneid, "SYMBOL", "ENTREZID", org.Hs.eg.db)


#pathway overrepresentation analyses
kegg_df <- enrichKEGG(gene = entrez_ids$ENTREZID,
                      organism = "hsa",
                      keyType = "kegg",
                      # OrgDb = org.Hs.eg.db, 
                      #ont = "MF", 
                      pAdjustMethod = "BH", 
                      qvalueCutoff = 0.05)


barplot(kegg_df, showCategory = 15, title = "10 most enriched pathways in co-expressed protein coding genes among trained indivuduals at postexercise")+
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8) , plot.title = element_text(hjust = 0.5, size = 8))
```

#### Volume-specific benefits

The modality and/or intensity of exercise training affects the responses . That is, there is a dose-dependent response to RT @hammarstrom_benefits_2020. This suggests there should be a difference between the individual legs that performed relatively more intense exercise than those that performed less.

##### Non-trained leg as a reference level

Using the control group as reference level, 16 and 18 lncRNAs were differentially expressed at post and mid exercise respectively. The interaction between set 6 and time gave 23 DE lncRNAs at mid exercise and 11 at postexercise. 13 lncRNAs were differentially expressed in set3 at mid exercise while 10 were differentially expressed at post exercise. A volcano plot showing the DE lncs are shown in the figure below.

```{r, zero_reference, echo=FALSE}
Vol <- readRDS("data/seqwrap_generated_models/filtered_volume_model.RDS")

vol_0_set6_t3 <- Vol %>%
  dplyr::filter(coef == "conditionset6:timet3")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)

vol_0_set6_t4 <- Vol %>%
  dplyr::filter(coef == "conditionset6:timet4")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)



vol_0_set3_t3 <- Vol %>%
  dplyr::filter(coef == "conditionset3:timet3")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)

vol_0_set3_t4 <- Vol %>%
  dplyr::filter(coef == "conditionset3:timet4")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)


 plot_volcano(vol_0_set6_t3, "DE lncRNAs set 6  mid exercise set zero reference ")


 plot_volcano(vol_0_set6_t4, "DE lncRNAs set 6  post exercise set zero reference ")



 plot_volcano(vol_0_set3_t3, "DE lncRNAs set 3  mid exercise set zero reference ")



 plot_volcano(vol_0_set3_t4, "DE lncRNAs set 3  post exercise set zero reference ")

```

These DE lncRNAs are somewhat similar to those in the trained versus untrained analyses. Except that the trained were seperated into the two different RT groups and compared to the untrained.

The 13 DE lncRNAs of set3 at midexercise coexpressed with 749 protein-coding genes and 745 at midexercise.

The 11 DE lncRNA of set6 at postexercise had 1077 coexpressed protein coding genes and 1114 at midexercise

```{r, set6_correlation_midexc, echo=FALSE}
vol6_df <- readRDS("data/seqwrap_generated_models/volume_coexpression_models/set6_midexc_counts.RDS")

cor_df <- enrichGO(gene = vol6_df$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "cc",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(cor_df, showCategory = 15,

        font.size = 5, title = "15 top cellular components of coexpressed proteins at  midexercise of set6 legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8))



cor_df <- enrichGO(gene = vol6_df$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "MF",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(cor_df, showCategory = 15,

        font.size = 5, title = "15 top molecular functions of coexpressed proteins at  midexercise of set6 legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8))


cor_df <- enrichGO(gene = vol6_df$geneid,
                   universe = genes_fpkm$gene_name,
                   keyType = "SYMBOL",
                   OrgDb = org.Hs.eg.db,
                   ont = "BP",
                   pAdjustMethod = "BH",
                   qvalueCutoff = 0.05,
                   readable = T)

dotplot(cor_df, showCategory = 15,

        font.size = 5, title = "15 top biological processes of coexpressed proteins at  midexercise of set6 legs") +
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8))



entrez_ids <- bitr(vol6_df$geneid, "SYMBOL", "ENTREZID", org.Hs.eg.db)


#pathway overrepresentation analyses
kegg_df <- enrichKEGG(gene = entrez_ids$ENTREZID,
                      organism = "hsa",
                      keyType = "kegg",
                      # OrgDb = org.Hs.eg.db,
                      #ont = "MF",
                      pAdjustMethod = "BH",
                      qvalueCutoff = 0.05)


barplot(kegg_df, showCategory = 15, title = "10 most enriched pathways in co-expressed protein coding genes among set6 legs at midexercise")+
  theme(axis.text = element_text(size = 8), axis.text.y = element_text(size = 8), axis.title.x = element_text(size = 8))

```

##### The difference between 3 sets and 6 sets of RT

To ascertain the difference in both RT intensities, the 3 set samples were used as reference level in a model. This gave the DE lncRNAs in the 6 sets samples in comparison to the 3 sets, in addition to other coefficients. There were 15 and 14 DE lncRNAs at post and mid exercise respectively. See table below.

```{r, set_3_base, echo=FALSE}
vol_3_base <- readRDS("data/seqwrap_generated_models/filtered_vol_model_withset3_baseline.RDS")


set6_t4_vol3_base <- vol_3_base %>%
  dplyr::filter(coef == "conditionset6:timet4") %>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)

set6_t3_vol3_base <- vol_3_base %>%
  dplyr::filter(coef == "conditionset6:timet3") %>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)


set6_t4_vol3_base_plot <- plot_volcano(set6_t4_vol3_base, "DE lncRNAs in set6 versus set3 at postexercise")

set6_t3_vol3_base_plot <- plot_volcano(set6_t3_vol3_base, "DE lncRNAs in set 6 versus set 3 at midexercise ")

grid.arrange(set6_t3_vol3_base_plot, set6_t4_vol3_base_plot)
```

#### Potential functional characterization of the differentially expressed lncRNAs

There is a poor understanding of the lncRNA functions owing to their relative low expressions , weak conservation and relatively less research into them @uszczynska-ratajczak_towards_2018. The most common method used in functional characterization of lncRNAs is the "guilty by association" method of inferring the functions of lncRNAs by the function of protein-coding genes with correlating expression patterns [@bonilauri2020; @chapman2022, @seifuddin2020]. This method assumes that expression patterns encode functionality @uszczynska-ratajczak_towards_2018.

The large number of correlating expressions in protein-coding genes needs experimental validation, though this tends to agree with @correia2021 that a particular lncRNA could be regulating different cellular processes and thus might be co-expressing with different protein-coding genes. Worthy of mention is the fact that correlation does not necessarily mean causation ; or regulation in this case. Though the functions of lncRNAs should be studied using wet-lab based methods, the increasing knowledge and access to computational methods for gene expression studies makes it easier to begin somewhere, by filtering a few lncRNAs from the thousands from where to begin the studies.

While this study does not enforce predictions of the functional roles of lncRNAs, it provides pointers towards lncRNAs of interest and perhaps points to areas where we could further enhance the understanding of the regulatory landscape of PRET and its benefits.

## Conclusion

The roles of lncRNAs should be explored beyond diseases, this could not only be useful towards personalisation of RT , but may also shed light on how RT can be used for health promotion and/or disease prevention . Understanding how RT impacts lncRNAs that are biomarkers of diseases could further the possibility of using RT for therapeutic interventions. While lncRNAs are believed to be regulators of different aspects of cellular physiology, it might also be helpful to investigate the regulation of these "regulators". lncRNA expression is said to be tissue specific, which suggests that the results obtained in this study might be specific to the skeletal muscle, but the benefits of RT are not limited to the muscles alone, it might thus be important for further studies to investigate if the benefits of RT as regards lncRNAs extend beyond skeletal muscles.

RT leads to the change in expression of certain lncRNAs overtime, with the specific RT condition mediating specific expression patterns. These computationally derived results will need to be validated using wet-lab methods. To validate the functional roles of these DE lncRNAs, *in vitro* studies involving their manipulation is recommended. Also, it might be pertinent to study them as they relate to muscle strength and/of muscle gain, both of which are measurable impacts of RT

# References

::: {#refs}
:::

:::

# 

# Appendix
