---
title: "Effects of progressive resistance training on the expression of long non-coding RNAs in skeletal muscle of young adults"
format: docx
output:
  bookdown::word_document2: default
  bookdown::pdf_documents: default
editor: visual
csl: science.csl
bibliography: My_Library.bib
target_journal: https://www.tandfonline.com/journals/krnb20
prefer-html: true
---

## Abstract 

## Introduction

The skeletal muscle is important for physical health and vitality. It responds to both use and disuse by hypertrophying, and loss of muscle strength and mass respectively [@lim_evidence-based_2022; @khan_increased_2020 ]. Human age correlates to loss of muscle mass and strength [@power_human_2013; @volpi_muscle_2004 ] which in turn correlates to the risks of falls, injuries and impaired mobility @wolfson_strength_1995 . Certain disease conditions like chronic obstructive pulmonary disease (COPD) could also accelerate loss in muscle mass and strength @molmen_chronic_2021 . Thus, muscle strength and mass are important for improved quality of life.

Progressive resistance exercise training (RT); a type of exercise where the skeletal muscle is exercised against progressively increased types of resistance @liu_progressive_2009 such as free weights , is the most potent non-pharmacological method of stimulating muscle hypertrophy and countering the loss in muscle strength and mass [@lim_evidence-based_2022; @hurst_resistance_2022] . RT has been shown to reduce blood glucose levels [@xie_effects_2022; @westcott_resistance_2012] improve strength and performance in older adults @liu_progressive_2009 and in children with celebral palsy @hanssen_progressive_2022 , improve health status and pain intensity in women with fibromyalgia [@larsson_resistance_2015; @busch_resistance_2013] , counteract the adverse effects of anrogen deprivation therapy in prostate cancer patients @houben_resistance_2023 , as well as decrease menopause-related symptoms @berin_resistance_2019. The benefits of RT extend beyond physical health to include the mental well being of participants @kekalainen_effects_2018 and correlate to the volume of RT done @hammarstrom_benefits_2020 . Besides volume, factors such as ribosome biogenesis, transcriptome profile and responses, metabolites profile etc determine an individual's response to RT [@lim_evidence-based_2022; @hammarstrom_ribosome_2022].

Numerous studies have reported the gene expression patterns following RT, but how these patterns are regulated is poorly understood. Understanding these could also help understand the factors regulating response to RT. Long non-coding RNAs (lncRNAs) are said to regulate the expression of protein-coding genes [@statello_gene_2021; @doi:10.1177/2472630318780639] as well as different aspects of cellular physiology and function [@oo_long_2022; @bridges_lnccation_2021] . lncRNAs are non-protein-coding RNAs of more than 200 nucleotides in length. Most lncRNAs have low levels of expression and sequence conservation @mattick_long_2023 , their expressions however are highly tissue and condition-specific [@statello_gene_2021; @bonilauri_long_2020; @ali_beyond_2020] . Over 240,000 lncRNAs in humans have been curated @10.1093/nar/gky960 even though not much is known about the functional roles of a majority of them @ali_beyond_2020 . Some have been reported to play roles in cell cycle regulation @kitagawa_cell_2013 , differentiation @delas_lncrna_2017 , metabolism @sirey_long_2019 , and muscle regeneration @sweta_importance_2019 . There is currently significant uncertainty about the mechanisms of actions of most lncRNAs. Figuring out lncRNAs that are differentially expressed in a given condition could thus be the first in a series of steps aimed towards elucidating their functions and mechanisms of action.

Most publicly available research on this group of RNAs are focused on their roles in diseases, especially cancers, . Relatively much less like @bonilauri_long_2020 are available on their roles in health promotion. It perhaps pertinent to investigate the possible functional roles of lncRNAs in health promoting activities like RT. This knowledge could not only contribute towards a deeper understanding of the mechanisms of response to RT, but perhaps optimizing the benefits as well as personalizing an individual's RT regimen.

This study aims to identify lncRNAs that are differentially expressed following RT based on exercise conditions, that is, between the trained and untrained, and between RT volumes among the trained. The study hopes to enhance the current knowledge about the responses to RT and how these can be optimized and personalized for individual participants.

## Methods

### Study design, RNA extraction and sequencing 

Study design , sample preparation and RNA sequencing were done as described by Hamarsland et al (in_view).

### Data Analyses

The quality of the RNA sequences was checked using FastQc (v0.11.5) @andrews2012 while RSEM (v.1.3.3-foss-2019b) @li2011 was used to align and count the reads mapping to human genome(GRCh 38) with gencode v40 primary assembly annotation . The expected count and FPKM values of the gene level counts were used during the data analyses . Further data extraction and statistical analyses were done on RStudio (2023.06.2Â Build 561). An overview of data analyses steps is shown in Figure 1

#### Models and visualization

Using BiomaRt @durinck_mapping_2009 , all genes annotated as lncRNA by the Ensemble database were filtered for downstream analyses. A total of 1024 lncRNAs were included in the analyses. These lncRNA counts were modeled using negative binomial GLMM (Generalized Linear Mixed Model) with the effective library size and time as fixed effects. Each participant was used as random effect having an individual intercept. To analyse the effect of RT over time, the trained legs were regarded as one and differential expression of lncRNA between the trained and the untrained contralateral leg was evaluated with the interaction between training status and time added to the fixed effects. To analyse the volume-dependent effects of RT on lncRNA expression, the interaction between the different training conditions and time were added to the fixed effects. Both model types were iteratively fitted using glmmTMB in Seqwrap , an R package built by Hammastrom et al (in view). All models were evaluated for uniformity and dispersion using DHARma (0.4.6). lncRNAs were identified as differentially expressed when their absolute log2 fold change was greater than 1 or less than -1, their adjusted p-values equal to or less than 0.05, and the p values of their model's uniformity and dispersion test above 0.05

####  Coexpression analyses

To understand the potential functions of the differentially expressed lncRNAs, coexpressed protein-coding genes were modeled and iteratively fitted using lmer @bates_fitting_2015 in Seqwrap . Each lncRNA count, time, condition, and sex of participants were used as fixed effects, while each participant was used as random effect with a fixed intercept. Coexpressed proteins were identified as those with adjusted p values aboe or at 0.05 and p values of their DHaRma uniformity test above 0.05. Gene ontology and KEGG gene sets of the coexpressed protein-coding genes were retrieved from org.Hs.eg.db

```{r Steps_analyses, fig.cap="Flow diagram sowing steps in analysis", fig.align = "right", echo=FALSE}

DiagrammeR::grViz("digraph {
  graph [layout = dot, rankdir = TB]
  
  node [shape = rectangle]        
  rec1 [label = 'Quality check (FastQC)']
  rec2 [label = 'Mapping to Human genome (GRCh 38)']
  rec3 [label =  'Read Counts (RSEM)']
  rec4 [label = 'Differential lncRNA expression analyses using GLMM']
  rec5 [label = 'Correlation analyses using lmer']
  rec6 [label = 'Gene set enrichment analyses']

  
  # edge definitions with the node IDs
  rec1 -> rec2 -> rec3 -> rec4 -> rec5 -> rec6
  }",
  height = 500)

```

```         
```

```         
```

## Results and Discussion

#### Trained versus untrained

15 lncRNAs were differentially expressed between the trained and untrained legs at midexercise, while only one lncRNA was differentially expressed by the end of the training regimen.

Using our method of finding correlations between the expression of lncs and protein-coding genes,

```{r, DE midexercise, echo=FALSE}

df <- readRDS("/data/seqwrap_generated_models/filtered_training_model.RDS")
t3 <- df %>%
  dplyr::filter(coef == "training_statustrained:timet3")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1) %>%
  select(target,adj.p, log2fc, pval.unif, pval.disp )
knitr::kable(t3, caption ="Differentially expressed lncRNAs halfway through the training program" )
```

```{r,DE postexercise,  echo=FALSE}

t4 <- df %>%
  dplyr::filter(coef == "training_statustrained:timet4")%>%
  dplyr::filter(log2fc >= 1 | log2fc <= -1)%>%
  select(target,adj.p, log2fc, pval.unif, pval.disp )
knitr::kable(t4, caption = "DE lncRNAs between the trained and untrained legs postexercise" )
```

DUXAP9 was downregulated in the trained leg postexercise. Its overexpression has been linked to different cancers [@zhu2018; @chen2019; @zhu2019] . DUXAP9's expression according to the model, correlated with the expression of 184 protein-coding genes.

The 15 lncRNAs at mid exercise correlated to 719 protein coding genes

#### Volume specific benefits

14 and 15 lncRNAs were differentially expressed at post and mid exercise respectively between the legs performed 3 sets and 6 sets of RT respectively

There is a poor understanding of the lncRNA functions owing to their relative low expressions , weak conservation and the relative less research into them @uszczynska-ratajczak_towards_2018

lncRNAs act as regulators of different mechanisms in the cell cycle @kitagawa_cell_2013 , @khan_distinct_2023

correlations in gene expression are used to infer the functions of genes @chapman_correlated_2022 , @seifuddin_lncrnakb_2020 . This guilt by association method assumes that expression patterns encode functionality @uszczynska-ratajczak_towards_2018

lncRNA research would most likely help understanding the underlying mechanisms that influence the benefits accrued from PRET

functionality of lncRNAs should be studied using wet-lab based methods. The increasing knowledge and access to computational methods for gene expression studies makes it easier to begin somewhere, by filtering a few lncRNAs from the thousands fom where to begin the studies.

While this study does not enforce predictions of the functional roles of lncRNAs, it provides pointers towards lncRNAs of interest and perhaps points to areas where we could further enhance the understanding of the regulatory landscape of PRET and its benefits

# References

::: {#refs}
:::
:::

# 

# Appendix
