---
title: "Effects of progressive resistance training on the expression of long non-coding RNAs in skeletal muscle of young adults"
format: docx
output:
  bookdown::word_document2: default
  bookdown::pdf_documents: default
editor: visual
csl: science.csl
bibliography: My_Library.bib
target journal :https://www.tandfonline.com/journals/krnb20
prefer-html: true
---

## Abstract 

## Introduction

The skeletal muscle is important for physical health and vitality. It responds to both use and disuse by hypertrophying, and loss of muscle strength and mass respectively [@lim_evidence-based_2022; @khan_increased_2020 ]. Human age correlates to loss of muscle mass and strength [@power_human_2013; @volpi_muscle_2004 ] which in turn correlates to the risks of falls, injuries and impaired mobility @wolfson_strength_1995 . Certain disease conditions like chronic obstructive pulmonary disease (COPD) could also accelerate loss in muscle mass and strength @molmen_chronic_2021 . Thus, muscle strength and mass are important for improved quality of life.

Progressive resistance exercise training (RT); a type of exercise where the skeletal muscle is exercised against progressively increased types of resistance @liu_progressive_2009 such as free weights , is the most potent non-pharmacological method of stimulating muscle hypertrophy and countering the loss in muscle strength and mass [@lim_evidence-based_2022; @hurst_resistance_2022] . RT has been shown to reduce blood glucose levels [@xie_effects_2022; @westcott_resistance_2012] improve strength and performance in older adults @liu_progressive_2009 and in children with celebral palsy @hanssen_progressive_2022 , improve health status and pain intensity in women with fibromyalgia [@larsson_resistance_2015; @busch_resistance_2013] , counteract the adverse effects of anrogen deprivation therapy in prostate cancer patients @houben_resistance_2023 , as well as decrease menopause-related symptoms @berin_resistance_2019. The benefits of RT extend beyond physical health to include the mental well being of participants @kekalainen_effects_2018 and correlate to the volume of RT done @hammarstrom_benefits_2020 . Besides volume, factors such as ribosome biogenesis, transcriptome profile and responses, metabolites profile etc determine an individual's response to RT [@lim_evidence-based_2022; @hammarstrom_ribosome_2022].

Numerous studies have reported the gene expression patterns following RT, but how these patterns are regulated is poorly understood. Understanding these could also help understand the factors regulating response to RT. Long non-coding RNAs (lncRNAs) are said regulate the expression of protein-coding genes [@statello_gene_2021; @doi:10.1177/2472630318780639] as well as different aspects of cellular physiology and function [@oo_long_2022; @bridges_lnccation_2021] . lncRNAs are non-protein-coding RNAs of more than 200 nucleotides in length. Most lncRNAs have low levels of expression and sequence conservation @mattick_long_2023 , their expressions however are highly tissue and condition-specific [@statello_gene_2021; @bonilauri_long_2020; @ali_beyond_2020] . Over 240,000 lncRNAs in humans have been curated @10.1093/nar/gky960 even though not much is known about the functional roles of a majority of them @ali_beyond_2020 . Some have been reported to play roles in cell cycle regulation @kitagawa_cell_2013 , differentiation @delas_lncrna_2017 , metabolism @sirey_long_2019 , and muscle regeneration @sweta_importance_2019 . There is currently significant uncertainty about the mechanisms of actions of most lncRNAs. Figuring out lncRNAs that are differentially expressed in a given condition could thus be the first in a series of steps aimed towards elucidating their mechanisms of action.

Most publicly available research on this group of RNAs are focused on their roles in diseases, especially cancers, . Relatively much less like @bonilauri_long_2020 are available on their roles in health promotion. It perhaps pertinent to investigate the possible functional roles of lncRNAs in health promoting activities like RT. This knowledge could not only contribute towards a deeper understanding of the mechanisms of response to RT, but perhaps optimizing the benefits as well as personalizing an individual's RT regimen.

This study aims to identify lncRNAs that are differentially expressed following RT based on exercise conditions, that is, between the trained and untrained and between volumes among the trained. It will enhance the current knowledge about the responses to PRET and how these can be optimized and personalized for individual participants

## Methods

### Study design, RNA extraction and sequencing 

Study design , sample preparation and RNA sequencing were done as described by Hamarsland et al (in_view).

### Data Analyses

The quality of the RNA sequences was checked using FastQc (v0.11.5) @andrews2012 while RSEM (v.1.3.3-foss-2019b) @li2011 was used to align and count the reads mapping to human genome(GRCh 38) with gencode v40 primary assembly annotation . Further data extraction and statistical analyses were done on RStudio (2023.06.2Â Build 561) . The expected count and FPKM values of the gene level counts were used during the data analyses . An overview of data analyses steps is shown in Figure 1

#### Models and visualization

using BiomaRt @durinck_mapping_2009 , all genes annotated as lncRNA by the Ensemble database were filtered for downstream analyses. To compare the influence of the effective library size of all the genes, versus only those annotated as lncRNAs, two types of negative binomial regression models were built using Seqwrap, an R package developed in-house, one normalized using lncRNAs alone, and another using the full genes counts.

Two models were built, one to determine the effect of training by grouping the participants' legs into trained and untrained. The other model was designed to investigate the effect of the different exercise conditions , that is set 6, set 3 and set 0. lncRNAs with p values less than or below 0.05, and log fold 2 change above 0.5 were extracted

####  Coexpression analyses

To understand the potential functions of the differentially expressed lncRNAs, a model was built using lmer @bates_fitting_2015 to calculate the correlation between each individual lncRNA expression and the expression of the genes annotated by Ensemble as "protein coding genes" . This model built to identify protein-coding genes with similar expression patterns as the lncRNAs given the condition and time of PRET .

```{r Steps_analyses, fig.cap="Flow diagram sowing steps in analysis", fig.align = "right", echo=FALSE}

DiagrammeR::grViz("digraph {
  graph [layout = dot, rankdir = TB]
  
  node [shape = rectangle]        
  rec1 [label = 'Quality check (FastQC)']
  rec2 [label = 'Mapping to Human genome (GRCh 38)']
  rec3 [label =  'Read Counts (RSEM)']
  rec4 [label = 'Differential lncRNA expression analyses using GLMM']
  rec5 [label = 'Correlation analyses using lmer']
  rec6 [label = 'Gene set enrichment analyses']

  
  # edge definitions with the node IDs
  rec1 -> rec2 -> rec3 -> rec4 -> rec5 -> rec6
  }",
  height = 500)

```

```         
```

```         
```

## Results

### Discussion

Thereis a poor understanding of the lncRNA functions owing to their relative low expressions , weak conservation and the relative less research into them @uszczynska-ratajczak_towards_2018

The lncRNA expression of biopsies from the vastus lateralis muscle of young individuals who participated in the Contratrain training exercise were compared and analysed to detect their differential expression based on exercise condition and time, as well as the mRNA genes possibly regulated by the lncRNAs

lncRNAs act as regulators of different mechanisms in the cell cycle @kitagawa_cell_2013 , @khan_distinct_2023

correlations in gene expression are used to infer the functions of genes @chapman_correlated_2022 , @seifuddin_lncrnakb_2020 . This guilt by association method assumes that expression patterns encode functionality @uszczynska-ratajczak_towards_2018

lncRNA research would most likely help understanding the underlying mechanisms that influence the benefits accrued from PRET

functionality of lncRNAs should be studied using wet-lab based methods. The increasing knowledge and access to computational methods for gene expression studies makes it easier to begin somewhere, by filtering a few lncRNAs from the thousands fom where to begin the studies.

While this study does not enforce predictions of the functional roles of lncRNAs, it provides pointers towards lncRNAs of interest and perhaps points to areas where we could further enhance the understanding of the regulatory landscape of PRET and its benefits

# References

::: {#refs}
:::
:::

# 

# Appendix
